src/main/java/uk/gov/laa/ccms/caab/controller/notifications/ActionsAndNotificationsController.java

```diff
@@ -0,0 +1,220 @@
+
+    return "notifications/actions-and-notifications-search";
+  }
+
+  /**
+   * Processes the search form from the Notifications Search page.
+   *
+   * @param user          current user details.
+   * @param criteria      the search criteria object in the model.
+   * @param model         the model.
+   * @param bindingResult Validation result of the search criteria form.
+   * @return Either redirects to the search results or reloads the form with validation errors.
+   */
+  @PostMapping("/notifications/search")
+  public String notificationsSearch(
+      @ModelAttribute(USER_DETAILS) UserDetail user,
+      @ModelAttribute(NOTIFICATION_SEARCH_CRITERIA) NotificationSearchCriteria criteria,
+      Model model, BindingResult bindingResult) {
+
+    notificationSearchValidator.validate(criteria, bindingResult);
+    if (bindingResult.hasErrors()) {
+      populateDropdowns(user, model, criteria);
+      return "notifications/actions-and-notifications-search";
+    }
+
+    return "redirect:/notifications/search-results";
+  }
+
+  private void populateDropdowns(UserDetail user, Model model,
+      NotificationSearchCriteria criteria) {
+    Mono<List<ContactDetail>> feeEarners = providerService.getProvider(user.getProvider().getId())
+        .map(providerService::getAllFeeEarners);
+    // get the notification types
+    Mono<CommonLookupDetail> notificationTypes = commonLookupService.getNotificationTypes();
+    // get the Users
+    Mono<UserDetails> users = userService.getUsers(user.getProvider().getId());
+
+    // Zip all Monos and populate the model once all results are available
+    Mono.zip(feeEarners, notificationTypes, users)
+        .doOnNext(tuple -> {
+          model.addAttribute("feeEarners", tuple.getT1());
+          model.addAttribute("notificationTypes", tuple.getT2().getContent());
+          model.addAttribute("users", tuple.getT3().getContent());
+        })
+        .block();
+    model.addAttribute("criteria", criteria);
+
+  }
+
+  private static void populateModelWithDefaultValues(Model model) {
+    model.addAttribute(SORT_FIELD, "assignDate");
+    model.addAttribute(SORT_DIRECTION, "asc");
+    model.addAttribute(REVERSE_SORT_DIRECTION, "desc");
+  }
+
+  private static void populateCriteria(String sort,
+      String notificationType, UserDetail user,
+      NotificationSearchCriteria criteria) {
+    criteria.setLoginId(user.getLoginId());
+    criteria.setUserType(user.getUserType());
+    criteria.setAssignedToUserId(user.getLoginId());
+    criteria.setNotificationType(notificationType);
+    if (StringUtils.isNotEmpty(notificationType) && notificationType.equals("all")) {
+      // do not set a notification type and reset the search criteria
+      resetCriteria(criteria);
+    }
+    // Get the sort parameters
+    if (StringUtils.isNotEmpty(sort)) {
+      criteria.setSort(sort);
+    }
+  }
+
+  private static void resetCriteria(NotificationSearchCriteria criteria) {
```

Comment:
can this be moved to the NotificationSearchCriteria as a constructor?

Reply:
yes.  Will create new object.